api/src/main/java/org/openmrs/api/handler/PatientDataUnvoidHandler.java

```diff
@@ -68,14 +68,19 @@ public void handle(Patient patient, User originalVoidingUser, Date origParentVoi
 			if (CollectionUtils.isNotEmpty(orders)) {
 				for (Order order : orders) {
 					if (order.isVoided() && order.getDateVoided().equals(origParentVoidedDate)
-					        && order.getVoidedBy().equals(originalVoidingUser)) {
+							&& order.getVoidedBy().equals(originalVoidingUser)) {
 						os.unvoidOrder(order);
 					}
 				}
 			}
-
+			
 			CohortService cs = Context.getCohortService();
-			cs.patientUnvoided(patient, patient.getVoidedBy(), patient.getDateVoided(), patient.getVoidReason());
+			if (patient.getId() != null) {
+				if (patient.getDateVoided().equals(origParentVoidedDate) &&
+						patient.getVoidedBy().equals(originalVoidingUser)) {
```

Comment:
For the condition to be in patientUnvoided(), shouldn't the patient be "unvoided" at the time? With the fields, dateVoided, voidedBy == null? This falls back to my earlier question about the unvoiding logic: [here](https://github.com/openmrs/openmrs-core/commit/0b662c88e8352da41bb3ac6525557b4801bf3e48#commitcomment-21178110). I will probably test again and get back to this comment.